$NetBSD$

The task-controller and jsvc components only build on Linux. The librecordio
component doesn't like to build against xerces-c (recent GCC too aggressive?).
Simplify installation phase for pkgsrc purposes. Use proper prefix for Snappy.

--- build.xml.orig	2013-01-31 02:06:05.000000000 +0000
+++ build.xml
@@ -215,7 +215,7 @@
   <property name="bundle.snappy" value="false"/>
 
   <!-- Snappy native library location -->
-  <property name="snappy.prefix" value="/usr/local"/>
+  <property name="snappy.prefix" value="@BUILDLINK_PREFIX.snappy@"/>
   <property name="snappy.lib" value="${snappy.prefix}/lib"/>
   <property name="snappy.include" value="${snappy.prefix}/include"/>
 
@@ -1598,14 +1598,6 @@
       <fileset dir="${basedir}/.eclipse.templates"/>
     </copy>
 
-    <subant target="task-controller">
-      <fileset dir="." includes="build.xml"/>
-    </subant>
-
-    <subant target="jsvc">
-      <fileset dir="." includes="build.xml"/>
-    </subant>
-
     <chmod perm="ugo+x" type="file" parallel="false">
         <fileset dir="${dist.dir}/bin"/>
         <fileset dir="${dist.dir}/sbin"/>
@@ -1671,17 +1663,16 @@
     </macro_tar>
   </target>
 
-  <target name="bin-package" depends="compile, jar, examples, tools-jar, jar-test, ant-tasks, package-librecordio" 
+  <target name="bin-package" depends="compile, jar, examples, tools-jar, jar-test, ant-tasks" 
 		description="assembles platform-specific artifacts for binary targets">
     <delete dir="${dist.dir}"/>
     <mkdir dir="${dist.dir}"/>
     <mkdir dir="${dist.dir}/bin"/>
-    <mkdir dir="${dist.dir}/etc/hadoop"/>
     <mkdir dir="${dist.dir}/lib"/>
-    <mkdir dir="${dist.dir}/lib64"/>
     <mkdir dir="${dist.dir}/libexec"/>
     <mkdir dir="${dist.dir}/sbin"/>
     <mkdir dir="${dist.dir}/share/${name}/contrib"/>
+    <mkdir dir="${dist.dir}/share/${name}/examples"/>
     <mkdir dir="${dist.dir}/share/${name}/webapps"/>
     <mkdir dir="${dist.dir}/share/${name}/templates/conf"/>
     <mkdir dir="${dist.dir}/native"/>
@@ -1711,7 +1702,7 @@
     <exec dir="${dist.dir}" executable="sh" failonerror="true">
       <env key="BASE_NATIVE_LIB_DIR" value="${lib.dir}/native"/>
       <env key="BUILD_NATIVE_DIR" value="${build.dir}/native"/>
-      <env key="DIST_LIB_DIR" value="${dist.dir}/${lib.data.model}"/>
+      <env key="DIST_LIB_DIR" value="${dist.dir}/lib"/>
       <env key="NATIVE_PLATFORM" value="${build.platform}" />
       <arg line="${native.src.dir}/packageBinNativeHadoop.sh"/>
     </exec>
@@ -1748,7 +1739,7 @@
       </fileset>
     </copy>
 
-    <copy todir="${dist.dir}/etc/hadoop">
+    <copy todir="${dist.dir}/share/examples/hadoop">
       <fileset dir="${conf.dir}" excludes="**/*.template"/>
     </copy>
 
@@ -1770,19 +1761,16 @@
       </fileset>
     </copy>
 
-    <copy todir="${dist.dir}/${lib.data.model}" includeEmptyDirs="false" flatten="true">
+    <copy todir="${dist.dir}/lib" includeEmptyDirs="false" flatten="true">
       <fileset dir="${build.dir}/c++">
-        <include name="${build.platform}/lib/*"/>
+        <include name="${build.platform}/lib/libhadoop*.a"/>
+        <include name="${build.platform}/lib/libhdfs.la"/>
+        <include name="${build.platform}/lib/libhdfs.so.0.0.0"/>
       </fileset>
     </copy>
 
-    <subant target="task-controller">
-      <fileset dir="." includes="build.xml"/>
-    </subant>
-
-    <subant target="jsvc">
-      <fileset dir="." includes="build.xml"/>
-    </subant>
+    <symlink link="${dist.dir}/lib/libhdfs.so.0" resource="libhdfs.so.0.0.0"/>
+    <symlink link="${dist.dir}/lib/libhdfs.so" resource="libhdfs.so.0.0.0"/>
 
     <chmod perm="ugo+x" type="file" parallel="false">
         <fileset dir="${dist.dir}/bin"/>
